storage/statedb/cache_redis.go

```diff
@@ -76,22 +86,45 @@ func NewRedisCache(config TrieNodeCacheConfig) (*RedisCache, error) {
 		return nil, err
 	}
 
-	logger.Info("Initialize trie node cache with redis", "endpoint", config.RedisEndpoints,
+	cache := &RedisCache{
+		client:    cli,
+		setItemCh: make(chan setItem, redisSetItemChannelSize),
+		pubSub:    cli.Subscribe(),
+	}
+
+	workerNum := runtime.NumCPU()/2 + 1
+	for i := 0; i < workerNum; i++ {
+		go func() {
+			for item := range cache.setItemCh {
+				cache.set(item.key, item.value)
+			}
+		}()
+	}
+
+	logger.Info("Initialized trie node cache with redis", "endpoint", config.RedisEndpoints,
 		"isCluster", config.RedisClusterEnable)
-	return &RedisCache{client: cli, pubSub: nil}, nil
+	return cache, nil
 }
 
 func (cache *RedisCache) Get(k []byte) []byte {
 	val, err := cache.client.Get(hexutil.Encode(k)).Bytes()
 	if err != nil {
-		// TODO-Klyatn: Print specific errors if needed
 		logger.Debug("cannot get an item from redis cache", "err", err, "key", hexutil.Encode(k))
 		return nil
 	}
 	return val
 }
 
 func (cache *RedisCache) Set(k, v []byte) {
+	item := setItem{key: k, value: v}
+	select {
+	case cache.setItemCh <- item:
+	default:
+		logger.Error("redis setItem channel is full")
+	}
+}
+
```

Comment:
```suggestion

// set writes data synchronously. This can take some time to write all data.
// To write data asynchronously, please use Set instead.
```

Reply:
Thanks. I added the comments.